libflux/go/libflux/analyze.go

```diff
@@ -74,6 +77,35 @@ func Analyze(astPkg *ASTPkg) (*SemanticPkg, error) {
 	return p, nil
 }
 
+func FindVarType(astPkg *ASTPkg, varName string) (semantic.MonoType, error) {
+	defer func() {
+		// This is necessary because the ASTPkg returned from the libflux API calls has its finalizer
+		// set with the Go runtime. But the find_var_type() API will consume the AST package during
+		// the conversion from the AST package to the semantic package.
+		// Setting this ptr to nil will prevent a double-free error.
+		astPkg.ptr = nil
```

Comment:
Should we change the behavior of `find_var_type` to not consume the AST package? Does it need to become owner of the resource?